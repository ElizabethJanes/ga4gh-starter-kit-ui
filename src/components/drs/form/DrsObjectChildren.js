import React from 'react';
import DrsObjectRelatives from './DrsObjectRelatives';
import { 
    Typography
} from '@material-ui/core';

const DrsObjectChildren = props => {
    return (
        <DrsObjectRelatives
            header='Bundle Children'
            sectionDescription={
                <div>
                    <Typography variant='body2' align='left' color='textSecondary'>
                        This DRS Object is currently acting as a DRS Bundle. Bundles
                        contain references to multiple Child objects (single-blob DRS
                        Objects and/or DRS Bundles), enabling multiple DRS Objects to
                        be logically grouped in a nested structure. Only DRS Bundles
                        may have children, single-blob DRS Objects do not have
                        children.
                    </Typography>
                    <Typography variant='body2' align='left' color='textSecondary'>
                        The following listing displays all children for the current
                        DRS Bundle.
                    </Typography>
                </div>
            }
            relationship="children"
            objectName="child"
            activeDrsObject={props.activeDrsObject}
            relatives={props.children}
            addRelative={props.addChild}
            setRelativeId={props.setChildId}
            unsetRelativeValidity={props.unsetChildValidity}
            removeRelative={props.removeChild}
            retrieveDrsObject={props.retrieveDrsObject}
            readOnly={props.readOnly}
        />
    )
}

export default DrsObjectChildren;
