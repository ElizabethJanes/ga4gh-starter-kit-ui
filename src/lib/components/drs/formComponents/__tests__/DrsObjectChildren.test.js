import React from 'react';
import renderer from 'react-test-renderer';
import DrsObjectChildren from '../DrsObjectChildren';

let zeroDrsObjectChildren=[];
let multipleDrsObjectChildren=[
    {id: '123'},
    {id: '456'},
    {id: '789'}
]

test('SHOW <DrsObjectChildren /> should handle zero child objects', () => {
    const drsObjectChildren = renderer
    .create(<DrsObjectChildren 
    readOnly={true} 
    formViewType='SHOW'
    drs_object_children={zeroDrsObjectChildren} />)
    .toJSON();
    expect(drsObjectChildren).toMatchSnapshot();
});

/* test('SHOW <DrsObjectChildren /> should handle multiple child objects', () => {
    const drsObjectChildren = renderer
    .create(<DrsObjectChildren 
    readOnly={true} 
    formViewType='SHOW'
    drs_object_children={multipleDrsObjectChildren} />)
    .toJSON();
    expect(drsObjectChildren).toMatchSnapshot();
}); */

test('NEW <DrsObjectChildren /> should handle zero child objects', () => {
    const drsObjectChildren = renderer
    .create(<DrsObjectChildren 
    readOnly={false} 
    formViewType='NEW'
    drs_object_children={zeroDrsObjectChildren} />)
    .toJSON();
    expect(drsObjectChildren).toMatchSnapshot();
});

/* test('NEW <DrsObjectChildren /> should handle multiple child objects', () => {
    const drsObjectChildren = renderer
    .create(<DrsObjectChildren 
    readOnly={false} 
    formViewType='NEW'
    drs_object_children={multipleDrsObjectChildren} />)
    .toJSON();
    expect(drsObjectChildren).toMatchSnapshot();
}); */

test('EDIT <DrsObjectChildren /> should handle zero child objects', () => {
    const drsObjectChildren = renderer
    .create(<DrsObjectChildren 
    readOnly={false} 
    formViewType='EDIT'
    drs_object_children={zeroDrsObjectChildren} />)
    .toJSON();
    expect(drsObjectChildren).toMatchSnapshot();
});

/* test('EDIT <DrsObjectChildren /> should handle multiple child objects', () => {
    const drsObjectChildren = renderer
    .create(<DrsObjectChildren 
    readOnly={false} 
    formViewType='EDIT'
    drs_object_children={multipleDrsObjectChildren} />)
    .toJSON();
    expect(drsObjectChildren).toMatchSnapshot();
}); */