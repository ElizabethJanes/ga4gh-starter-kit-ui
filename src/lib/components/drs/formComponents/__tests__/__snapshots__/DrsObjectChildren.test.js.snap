// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EDIT <DrsObjectChildren /> should handle zero child objects 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Bundle Children
  </h6>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      This DRS Object is currently acting as a DRS Bundle. Bundles contain references to multiple Child objects (single-blob DRS Objects and/or DRS Bundles), enabling multiple DRS Objects to be logically grouped in a nested structure. Only DRS Bundles may have children, single-blob DRS Objects do not have children.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      The following listing displays all children for the current DRS Bundle.
    </p>
  </div>
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add child"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NEW <DrsObjectChildren /> should handle zero child objects 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Bundle Children
  </h6>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      This DRS Object is currently acting as a DRS Bundle. Bundles contain references to multiple Child objects (single-blob DRS Objects and/or DRS Bundles), enabling multiple DRS Objects to be logically grouped in a nested structure. Only DRS Bundles may have children, single-blob DRS Objects do not have children.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      The following listing displays all children for the current DRS Bundle.
    </p>
  </div>
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add child"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`SHOW <DrsObjectChildren /> should handle zero child objects 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Bundle Children
  </h6>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      This DRS Object is currently acting as a DRS Bundle. Bundles contain references to multiple Child objects (single-blob DRS Objects and/or DRS Bundles), enabling multiple DRS Objects to be logically grouped in a nested structure. Only DRS Bundles may have children, single-blob DRS Objects do not have children.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      The following listing displays all children for the current DRS Bundle.
    </p>
  </div>
  <div />
</div>
`;
